core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -37,6 +37,7 @@
  * @author Peter Karich
  */
 public class DataFlagEncoder extends AbstractFlagEncoder {
+
```

Comment:
The changes here look nice and reasonable to me - would you mind to add a unit test in combination with the GenericWeighting?  (not just the integration test)